modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/spaces/SpaceConnectedSitesModal.tsx

```diff
@@ -159,25 +152,30 @@ const SitesSelector = ({
 						{Liferay.Language.get('site')}
 					</label>
 
-					<Autocomplete
-						allowsCustomValue
-						id="siteSelector"
-						menuTrigger="focus"
-						onChange={setValue}
-						placeholder={Liferay.Language.get('select-a-site')}
-						value={value}
+					<ItemSelector<Site>
+						apiURL={`${location.origin}/o/headless-site/v1.0/sites?active=true`}
+						as={ClayInput}
+						items={site ? [site] : []}
+						onItemsChange={(items: Site[]) => {
+							if (items.length) {
+								setSite(items[0]);
+							}
+							else {
+								setSite(null);
+							}
+						}}
 					>
-						{sites.map((site) => (
-							<Autocomplete.Item
-								key={site.id}
-								onClick={() => {
-									setSiteSelected(site);
-								}}
+						{(item: Site) => (
+							<ItemSelector.Item
+								key={item.id}
+								textValue={item.name}
 							>
-								{getLocalizedName(site.name)}
-							</Autocomplete.Item>
-						))}
-					</Autocomplete>
+								<span className="inline-item inline-item-after">
```

Comment:
Why do we need these classes? We usually use these if we have an icon and some text together in an item.